### PR TITLE
DOCSP-34729: Update Manage Subscriptions page with Subscribe API for Flutter

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-34729

- [Manage Subscriptions - Flutter SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/docsp-34729-subscribe-api/sdk/flutter/sync/manage-sync-subscriptions/#subscribe-to-queries): Update page with new documentation for new .subscribe() and .unsubscribe() methods, similar to the other SDKs.

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-app-services work, if any

### Release Notes

- **Flutter** SDK
  - Sync Device Data/Manage Subscriptions: Update page to document the new Subscribe API with tested, Bluehawked code examples.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/dart/test/manage_sync_subscription_test.dart [hunk 1]
```diff
@@ -61,8 +61,8 @@ void main() {
       final planeQuery = realm.all<Plane>();
       final longTrainQuery = realm.all<Train>().query("numCars >= 4");
       realm.subscriptions.update((MutableSubscriptionSet mutableSubscriptions) {
-        mutableSubscriptions.add(planeQuery, name: "all-planes");
-        mutableSubscriptions.add(longTrainQuery, name: 'long-trains');
+        mutableSubscriptions.add(planeQuery, name: "all-planes"); // +1
```
Comment (krollins-mdb) [inline examples/dart/test/manage_sync_subscription_test.dart:64]: What are these numbers in comments for? I couldn't figure it out. 😅

File: examples/dart/test/manage_sync_subscription_test.dart [hunk 2]
```diff
@@ -136,22 +137,100 @@ void main() {
...
     });
     test('Remove all subscriptions', () async {
       // :snippet-start: remove-all-subscriptions
       realm.subscriptions.update((MutableSubscriptionSet mutableSubscriptions) {
         mutableSubscriptions.clear();
       });
       // :snippet-end:
+      expect(realm.subscriptions, isEmpty); // clear all
+    });
+    test('Add query with subscribe api', () async {
```
Comment (krollins-mdb) [inline examples/dart/test/manage_sync_subscription_test.dart:158]: I know the current tests don't have whitespace between them. Would you mind adding a space between `test` instances? It improves readability considerably.

File: source/sdk/flutter/sync/manage-sync-subscriptions.txt [hunk 3]
```diff
@@ -53,8 +61,120 @@ You subscription queries can either:
...
+Subscribe to Queries
+--------------------
+
+.. versionadded:: v1.6.0
+
+Flutter v1.6.0 adds experimental APIs that subscribe to and unsubscribe
+from a query's results. These APIs abstract away the details of
+manually adding and removing subscriptions.
+
+For all subscriptions, you need :an :ref:`authenticated user <flutter-authenticate>` and a :ref:`synced realm <flutter-open-synced-realm>`.
```
Comment (krollins-mdb) [inline source/sdk/flutter/sync/manage-sync-subscriptions.txt:75]:
```suggestion
For all subscriptions, you need an :ref:`authenticated user <flutter-authenticate>` and a :ref:`synced realm <flutter-open-synced-realm>`.
```

File: source/sdk/flutter/sync/manage-sync-subscriptions.txt [hunk 4]
```diff
@@ -53,8 +61,120 @@ You subscription queries can either:
...
+using the ``subscriptions`` API. Refer to the 
+:ref:`Performance Considerations <flutter-flex-sync-performance-considerations>` 
+section on this page for more information.
+
+Subscribe to a Query
+~~~~~~~~~~~~~~~~~~~~
+
+You can subscribe to the ``RealmResults`` of a query using the 
+:flutter-sdk:`subscribe() <realm/RealmResultsOfRealmObject/subscribe.html>` 
+method. When called, the The SDK creates the new subscription and adds it to the
```
Comment (krollins-mdb) [inline source/sdk/flutter/sync/manage-sync-subscriptions.txt:89]:
```suggestion
method. When called, the SDK creates the new subscription and adds it to the
```

File: source/sdk/flutter/sync/manage-sync-subscriptions.txt [hunk 5]
```diff
@@ -53,8 +61,120 @@ You subscription queries can either:
...
+overwrites the existing subscription.
+
+If you do not pass a subscription name, the name is set to ``null``, and 
+the subscription identifier is based on the query string. This means that 
+every time your query string changes, ``subscribe()`` creates a new subscription.
+
+To subscribe to a query, pass the following arguments to ``subscribe()``:
+
+- ``RealmResults query``: Required. A ``RealmResults`` object that you can create using the
+  :ref:`Realm Query Language query <rql>`.
```
Comment (krollins-mdb) [inline source/sdk/flutter/sync/manage-sync-subscriptions.txt:104]:
```suggestion
  :ref:`Realm Query Language <rql>`.
```

File: source/sdk/flutter/sync/manage-sync-subscriptions.txt [hunk 6]
```diff
@@ -53,8 +61,120 @@ You subscription queries can either:
...
+- ``String name``: Optional. Name for the subscription that you can refer to. 
+- ``bool update``:  Optional. When true, adding a subscription with an existing name
+  replaces the existing query with the new query. When false, the SDK
+  throws an exception for duplicate subscriptions. Only use with named subscriptions.
+
+In the following example, we subscribe to 2 new named queries.
```
Comment (krollins-mdb) [inline source/sdk/flutter/sync/manage-sync-subscriptions.txt:110]: The style guide calls for spelling out numbers zero through nine. https://www.mongodb.com/docs/meta/style-guide/style/numbers/#numbers-versus-words
```suggestion
In the following example, we subscribe to two new named queries.
```

File: source/sdk/flutter/sync/manage-sync-subscriptions.txt [hunk 7]
```diff
@@ -53,8 +61,120 @@ You subscription queries can either:
...
+   your subscriptions easier.
+
+Wait for a Query Subscription to Sync
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you subscribe to a query's results, the results do not contain objects
+until synced data is downloaded. When you do need to wait for synced objects to
+finish downloading, configure the :flutter-sdk:`waitForSyncMode <realm/WaitForSyncMode.html>` option. You can specify
+different behavior for your subscriptions and how they handle writing for
+downloads.
```
Comment (krollins-mdb) [inline source/sdk/flutter/sync/manage-sync-subscriptions.txt:128]: > how they handle writing for downloads

This bit could be confusing. I don't think we've equated sync download with writing data anywhere, though that's exactly what happens. I'm also not certain how this is different from the preceding "You can specify
different behavior...".

**Review by krollins-mdb — APPROVED** [review]
LGTM, with some minor suggestions. I know some of these comments are on things that aren't part of what you updated, but it can be hard to tell with large page changes.

File: examples/dart/test/manage_sync_subscription_test.dart (hunk 1 above)
Comment (cbullinger) [reply examples/dart/test/manage_sync_subscription_test.dart:64]: ha! they're for the assertions so I could track the number of subscriptions being added/subtracted per test. Probably unnecessary or I can update to be more meaningful

File: source/sdk/flutter/sync/manage-sync-subscriptions.txt (hunk 7 above)
Comment (cbullinger) [reply source/sdk/flutter/sync/manage-sync-subscriptions.txt:128]: This language was taken from the RN page. I can update it there page too.

Comment (krollins-mdb) [reply source/sdk/flutter/sync/manage-sync-subscriptions.txt:128]: lol. Thank you! I couldn't tell you if I wrote that sentence or it was part of the RN docs before I started working on them. 😅

**Pull request merged**